test/e2e/issues_test.go

```diff
@@ -264,6 +265,81 @@ var _ = Describe("Issues", func() {
 		// config will be in the failure log and we can see what's happened.
 		MustNotRun("kubectl get -oyaml hierarchyconfiguration hierarchy -n", nsParent)
 	})
+
+	It("Should propagate the change of source CRD - issue #53", func() {
+		// Create a parent namespace and a subnamespace for it.
+		CreateNamespace(nsParent)
+		MustRun("kubectl get ns", nsParent)
+		CreateSubnamespace(nsChild, nsParent)
+		crd := `# a simple CRD used for e2e testing only, should be deleted after finishing(or failing) this testcase.
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: eetests.e2e.hnc.x-k8s.io
+spec:
+  group: e2e.hnc.x-k8s.io
+  names:
+    kind: EETest
+    listKind: EETestList
+    plural: eetests
+    singular: eetest
+  scope: Namespaced
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        description: EETest is the Schema for the eetests API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: EETestSpec defines the desired state of EETest
+            properties:
+              foo:
+                description: Foo is an example field of EETest. Edit eetest_types.go
+                  to remove/update
+                type: string
+            type: object
+        type: object
+    served: true
+    storage: true`
+		// create CRD and set the propagation strategy for it.
+		MustApplyYAML(crd)
+		MustRun("kubectl hns config set-resource eetests --group e2e.hnc.x-k8s.io --mode Propagate --force")
```

Comment:
Can you undo this in `AfterEach`? Just run the same command again but set the mode to `Ignore` instead of `Propagate`.

Reply:
I find there will be a error condition appended in hncconfiguration, if it's okay?
```
    conditions:
    - lastTransitionTime: "1970-01-01T00:00:00Z"
      message: Resource "eetests.e2e.hnc.x-k8s.io" not found
      reason: ResourceNotFound
      status: "True"
      type: BadConfiguration
```

Reply:
Maybe run the command `kubectl hns config delete-type --group e2e.hnc.x-k8s.io --resource eetests` to delete this type is better?